pkg/logs/sender/sender.go

```diff
@@ -125,7 +127,10 @@ func (s *Sender) run() {
 		if s.senderDoneChan != nil {
 			senderDoneWg.Wait()
 			// In serverless ensure the payload is sent to all destinations to sync with a flush
-			s.flushDoneChan <- struct{}{}
+			// If there are multiple batch payloads, wait for all of them to be sent
+			if len(payload.Messages) < s.maxBufferSize {
+				go func() { s.flushDoneChan <- struct{}{} }()
+			}
```

Comment:
I don't have a better suggestion off the top of my head since I'm not super familiar with this code, but it does strike me that this is an imprecise way of determining whether we're sending multiple payloads (e.g. there could be one payload of exactly the maximum size). That might be a fine tradeoff if it's better than the current behavior, but again it might be worth a comment explaining why we're making that tradeoff and why we can't tell more precisely.